File: app_codebase/dart_bkend_base_02/bin/core/auth/verify_service_key.dart
/// Shared service key verification for HTTP guards and WS auth handshake.
library;

import '../errors/app_error.dart';
import '../errors/error_codes.dart';
import 'auth_config.dart';
import 'secret_compare.dart';

void verifyServiceKeyOrThrow(String? provided) {
  final expected = serviceKey();
  final value = (provided ?? '').trim();
  if (value.isEmpty || expected.isEmpty || !secretsEqual(value, expected)) {
    throw AppError(forbidden, message: 'Invalid service key');
  }
}

File: app_codebase/python_base_05/bin/core/auth/verify_service_key.py
"""Shared service key verification for HTTP guards and WS auth handshake."""

from __future__ import annotations

from core.auth.auth_config import service_key
from core.auth.secret_compare import secrets_equal
from core.errors.app_error import AppError
from core.errors.error_codes import FORBIDDEN


def verify_service_key_or_raise(provided: str | None) -> None:
    expected = service_key()
    value = (provided or "").strip()
    if not value or not expected or not secrets_equal(value, expected):
        raise AppError(FORBIDDEN, message="Invalid service key")
